ndk/src/performance_hint.rs

```diff
@@ -0,0 +1,326 @@
+//! Bindings for [`APerformanceHintManager`] and [`APerformanceHintSession`]
+//!
+//! `APerformanceHint` allows apps to create performance hint sessions for groups of
+//! threads, and provide hints to the system about the workload of those threads, to help the
+//! system more accurately allocate power for them. It is the NDK counterpart to the [Java
+//! PerformanceHintManager SDK API].
+//!
+//! [`APerformanceHintManager`]: https://developer.android.com/ndk/reference/group/a-performance-hint#aperformancehintmanager
+//! [`APerformanceHintSession`]: https://developer.android.com/ndk/reference/group/a-performance-hint#aperformancehintsession
+//! [Java PerformanceHintManager SDK API]: https://developer.android.com/reference/android/os/PerformanceHintManager
+#![cfg(feature = "api-level-33")]
+
+#[cfg(doc)]
+use std::io::ErrorKind;
+use std::{io::Result, ptr::NonNull, time::Duration};
+
+use crate::utils::status_to_io_result;
+
+/// An opaque type representing a handle to a performance hint manager.
+///
+/// To use:
+/// - Obtain the performance hint manager instance by calling [`PerformanceHintManager::new()`].
+/// - Create a [`PerformanceHintSession`] with [`PerformanceHintManager::new()`].
+/// - Get the preferred update rate with [`PerformanceHintManager::preferred_update_rate()`].
+#[derive(Debug, PartialEq, Eq, Hash)]
+#[doc(alias = "APerformanceHintManager")]
+pub struct PerformanceHintManager {
+    ptr: NonNull<ffi::APerformanceHintManager>,
+}
+
+impl PerformanceHintManager {
+    /// Retrieve a reference to the performance hint manager.
+    ///
+    /// Returns [`None`] on failure.
+    #[doc(alias = "APerformanceHint_getManager")]
+    pub fn new() -> Option<Self> {
+        NonNull::new(unsafe { ffi::APerformanceHint_getManager() }).map(|ptr| Self { ptr })
+    }
```

Comment:
As it turns out there was a hidden `static` keyword in this C function, this should always return the same singleton for the process:

https://issuetracker.google.com/issues/357214902#comment7

Meaning we can also derive `Send`/`Sync` and `Copy`/`Clone` since the user could anyway cheaply retrieve the same singleton from any thread.